src/popper/runner_host.py

```diff
@@ -193,7 +193,23 @@ def _create_container(self, cid, step):
         elif not self._config.skip_pull and not step.skip_pull:
             log.info(f"[{step.id}] docker pull {img}:{tag}")
             if not self._config.dry_run:
-                self._d.images.pull(repository=f"{img}:{tag}")
+                streamer = self._d.api.pull(
+                    repository=f"{img}:{tag}", decode=True,
+                )
+                
+                for chunk in streamer.splitlines():
+                    if self._config.quiet:
+                        continue
+                    chunk = chunk.strip()
+                    if chunk:
+                        import json
+                        chunk = json.loads(chunk)
+                        if 'id' in chunk:
+                            log.step_info(f"[{step.id}] " + chunk['id'] + ': ' + chunk['status'])
+                        else:
+                            log.step_info(f"[{step.id}] " + chunk['status'])
+                log.step_info(f"[{step.id}] {img}:{tag}")
```

Comment:
this line can be omitted, as the last line from docker pull is something like `Downloaded newer image for <url>:<tag>` so it is already informing about what got downloaded